src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -65,71 +64,142 @@
 
     private final StudentParticipationRepository studentParticipationRepository;
 
+    private final ObjectMapper objectMapper;
+
     private final FileService fileService;
 
     private final GitService gitService;
 
     private final ZipFileService zipFileService;
 
+    public static final String EXPORTED_EXERCISE_DETAILS_FILE_PREFIX = "Exercise-Details";
+
+    public static final String EXPORTED_EXERCISE_PROBLEM_STATEMENT_FILE_PREFIX = "Problem-Statement";
+
     public ProgrammingExerciseExportService(ProgrammingExerciseRepository programmingExerciseRepository, StudentParticipationRepository studentParticipationRepository,
-            FileService fileService, GitService gitService, ZipFileService zipFileService) {
+            FileService fileService, GitService gitService, ZipFileService zipFileService, MappingJackson2HttpMessageConverter springMvcJacksonConverter) {
         this.programmingExerciseRepository = programmingExerciseRepository;
         this.studentParticipationRepository = studentParticipationRepository;
+        this.objectMapper = springMvcJacksonConverter.getObjectMapper();
         this.fileService = fileService;
         this.gitService = gitService;
         this.zipFileService = zipFileService;
     }
 
     /**
-     * Export a programming exercise by creating a zip file. The zip file includes all student, template, solution,
-     * and tests repositories.
+     * Export programming exercise material for instructors including instructor repositories, problem statement (.md) and exercise detail (.json).
+     *
+     * @param exercise     the programming exercise
+     * @param exportErrors List of failures that occurred during the export
+     * @return the path to the zip file
+     */
+    public Path exportProgrammingExerciseInstructorMaterial(ProgrammingExercise exercise, List<String> exportErrors) {
```

Comment:
[optional] Since you have been introducing `Optional<File>` to the other methods, what do you think of wrapping the return type into `Optional<Path>`

Reply:
Thanks for your feedback. You are right, this is could be done for `exportProgrammingExerciseInstructorMaterial` and `exportProgrammingExerciseRepositories`. But these methods are used by another service as well. I think I would rather do that in a small follow-up PR to avoid that this PR entails further refactoring for example of the endpoint and related tests.